fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/DefaultLoanLifecycleStateMachine.java

```diff
@@ -42,6 +42,10 @@ public LoanStatus dryTransition(final LoanEvent loanEvent, final Loan loan) {
     public void transition(final LoanEvent loanEvent, final Loan loan) {
         LoanStatus newStatus = getNextStatus(loanEvent, loan);
         if (newStatus != null) {
+            if (isNotLoanCreation(loanEvent) && (loan.isClosed() || LoanStatus.OVERPAID.equals(loan.getStatus()))
```

Comment:
Please either use the loan methods to decide on the statuses or use the equality check everywhere, but do not mix them.
Example:

if (isNotLoanCreation(loanEvent) && (loan.isClosed() || **loan.isOverpaid())** && LoanStatus.ACTIVE.equals(newStatus))